File: app/src/main/java/com/ssy/ferry/listener/LooperObserver.java
package com.ssy.ferry.listener;

/**
 * 2019-10-21
 *
 * @author Mr.S
 */
public abstract class LooperObserver {

    public boolean isDispatchBegin = false;

  public  void dispatchBegin(long beginMs, long cpuBeginMs, long token) {
        isDispatchBegin = true;
    }

    void doFrame(
            String focusedActivityName,
            long start,
            long end,
            long frameCostMs,
            long inputCostNs,
            long animationCostNs,
            long traversalCostNs
    ) {

    }

    public  void dispatchEnd(
            long beginMs,
            long cpuBeginMs,
            long endMs,
            long cpuEndMs,
            long token,
            boolean isBelongFrame
    ) {
        isDispatchBegin = false;
    }

}

File: app/src/main/java/com/ssy/ferry/listener/LooperObserver.kt
package com.ssy.ferry.listener

/**
 * 2019-10-17
 * @author Mr.S
 */
open abstract class LooperObserver {

     var isDispatchBegin: Boolean = false

    open  fun dispatchBegin(beginMs: Long, cpuBeginMs: Long, token: Long) {
        isDispatchBegin = true
    }

    open   fun doFrame(
        focusedActivityName: String,
        start: Long,
        end: Long,
        frameCostMs: Long,
        inputCostNs: Long,
        animationCostNs: Long,
        traversalCostNs: Long
    ) {

    }

    open  fun dispatchEnd(
        beginMs: Long,
        cpuBeginMs: Long,
        endMs: Long,
        cpuEndMs: Long,
        token: Long,
        isBelongFrame: Boolean
    ) {
        isDispatchBegin = false
    }


}